teps/0140-producing-results-in-matrix.md

```diff
@@ -0,0 +1,624 @@
+        image: bash:latest
+        script: |
+          #!/usr/bin/env bash
+          echo -n "[\"linux\",\"mac\",\"windows\"]" | tee $(results.platforms.path)
+```
+
+`taskRun` created:
+
+```
+pr-get-platforms - [\"linux\",\"mac\",\"windows\"]
+```
+
+Now, a `matrix` when fanned out creates multiple `taskRuns` so for a `matrix` to emit results, it is unclear how
+each `results` would map back to the original matrixed `PipelineTask`. In the example below, the following `matrix` will
+produce 9 `taskRuns` that will each produce a string `result` named "report-url" for each combination of platform and
+browser. Since the order of the `taskRuns` is constant and deterministic, it will always produce `results` in the same
+order.
+
+```yaml
+Tasks:
+...
+kind: Task
+metadata:
+  name: task-producing-results
+spec:
+  params:
+    - name: platform
+    - name: browser
+  results:
+    - name: report-url
+  steps:
+    - name: produce-report-url
+      image: alpine
+      script: |
+        #!/usr/bin/env bash
+        echo "https://api.example/get-report/$(params.platform)-$(params.browser)" | tee $(results.report-url.path)
+---
+kind: PipelineRun
+metadata:
+  generateName: pipelinerun-with-matrix-task-
+spec:
+  serviceAccountName: "default"
+  pipelineSpec:
+    tasks:
+      - name: matrix-with-task-producing-results
+        matrix:
+          params:
+            - name: platform
+              value:
+                - linux
+                - mac
+                - windows
+            - name: browser
+              value:
+                - chrome
+                - safari
+                - firefox
+        taskRef:
+          name: task-producing-results
+          kind: Task
+```
+
+`taskRuns` Created - "report-url" result produced:
+
+```
+pr-produce-platforms-and-browsers-0 - "path/to/report/linux-chrome"
+pr-produce-platforms-and-browsers-1 - "path/to/report/linux-safari"
+pr-produce-platforms-and-browsers-2 - "path/to/report/linux-firefox"
+pr-produce-platforms-and-browsers-3 - "path/to/report/mac-chrome"
+pr-produce-platforms-and-browsers-4 - "path/to/report/mac-safari"
+pr-produce-platforms-and-browsers-5 - "path/to/report/mac-firefox"
+pr-produce-platforms-and-browsers-6 - "path/to/report/windows-chrome"
+pr-produce-platforms-and-browsers-7 - "path/to/report/windows-safari"
+pr-produce-platforms-and-browsers-8 - "path/to/report/windows-firefox"
+```
+
+### Goals
+
+The main goal of this TEP is to enable executing a `matrixed` `pipelineTask` that can emit `result` and support
+consuming all of the `results` that were produced in another `pipelineTask` as long as the type of `result` emitted
+was `string`.
+
+### Non-Goals
+
+The following are out of scope for this TEP:
+
+1. Support a matrixed `pipelineTask` that produces `result` of type `array` or `object` that are then consumed by
+   another `pipelineTask`.
+  - A matrixed `pipelineTask` can support `result` of any type as long as they aren’t consumed by
```

Comment:
I'm assuming they also can't be consumed in pipeline results, but you might be able to consume them with pipelines in pipelines or with a script?

Reply:
Yes, these can be consumed by `chains` and in future will be helpful in [consuming results from an individual instances](https://github.com/tektoncd/pipeline/issues/5266#issuecomment-1674089553) and could be useful while implementing pipelines in pipelines or with a scripts.